tests/test_ops.py

```diff
@@ -1,2 +1,61 @@
-""" Empty file for now
-"""
+import pytest
+
+from conftest import skipif
+
+from devito import Eq, Grid, Operator, TimeFunction, configuration  # noqa
+from devito.ops.node_factory import OPSNodeFactory
+from devito.ops.transformer import make_ops_ast
+from devito.symbolics import indexify
+
+
+pytestmark = skipif('noops')
+
+
+class TestOPSExpression(object):
+
+    @pytest.mark.parametrize('equation, expected', [
+        ('Eq(u,3*a - 4**a)', 'Eq(ut[OPS_ACC0(0)], -2.97015324253729)'),
+        ('Eq(u, u.dxl)',
+         'Eq(ut[OPS_ACC0(0)], -ut[OPS_ACC0(-1)]/h_x + ut[OPS_ACC0(0)]/h_x)'),
+        ('Eq(v,1)', 'Eq(vt[OPS_ACC0(0,0)], 1)'),
+        ('Eq(v,v.dxl + v.dxr - v.dyr - v.dyl)',
+         'Eq(vt[OPS_ACC0(0,0)], vt[OPS_ACC0(0,-1)]/h_y - vt[OPS_ACC0(0,1)]/h_y'
+            ' - vt[OPS_ACC0(-1,0)]/h_x + vt[OPS_ACC0(1,0)]/h_x)'),
+        ('Eq(v,v**2 - 3*v)',
+         'Eq(vt[OPS_ACC0(0,0)], vt[OPS_ACC0(0,0)]**2 - 3*vt[OPS_ACC0(0,0)])'),
+        ('Eq(v,a*v + b)',
+         'Eq(vt[OPS_ACC0(0,0)], 1.43*vt[OPS_ACC0(0,0)] + 9.87e-7)'),
+        ('Eq(v.dt,1/v + v**-1)',
+         'Eq(-vt[OPS_ACC1(0,0)]/dt + vt + 1[OPS_ACC0(0,0)]/dt, 2/vt[OPS_ACC1(0,0)])'),
+        ('Eq(w,c*w**2)',
+         'Eq(wt[OPS_ACC0(0,0,0)], 999999999999999*wt[OPS_ACC0(0,0,0)]**2)'),
+        ('Eq(w,u + v )', 'Eq(wt[OPS_ACC0(0,0,0)], ut[OPS_ACC1(0)] + vt[OPS_ACC2(0,0)])'),
```

Comment:
can you add some further tests with ``u.forward`` and ``v.forward``, so that we get to see ``ut0, ut1, ...`` ?

Reply:
I tried to added it, but u.forward, u.laplace, u.dx# ( wheren # > 1 ) doesn't seems to get translated with what i have. It seems there is something in Operator that I am not using that replace this u.forward for sympy symbols.

Reply:
My point is that this tests should be done when using `Operator`, but i can replicate the part of Operator that do this.